deployment/modules/gcp/key/main.tf

```diff
@@ -0,0 +1,60 @@
+terraform {
+  required_providers {
+    google = {
+      source  = "registry.terraform.io/hashicorp/google"
+      version = "6.1.0"
+    }
+  }
+}
+
+# Secret Manager
+
+# ECDSA key with P256 elliptic curve. Do NOT use this in production environment.
+#
+# Security Notice
+# The private key generated by this resource will be stored unencrypted in your 
+# Terraform state file. Use of this resource for production deployments is not 
+# recommended. Instead, generate a private key file outside of Terraform and 
```

Comment:
Nit - I'd remove the "instead...", let's not make security recommendations until we've figured out what we want to do :)

+1 for the Security notice, thanks!